src/main/java/org/opensearch/flowframework/util/WorkflowTimeoutUtility.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.util;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.ExceptionsHelper;
+import org.opensearch.client.Client;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.flowframework.transport.GetWorkflowStateAction;
+import org.opensearch.flowframework.transport.GetWorkflowStateRequest;
+import org.opensearch.flowframework.transport.WorkflowResponse;
+import org.opensearch.threadpool.Scheduler;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.opensearch.flowframework.common.CommonValue.PROVISION_WORKFLOW_THREAD_POOL;
+
+/**
+ * Utility class for managing timeout tasks in workflow execution.
+ * This class provides methods to schedule timeout handlers, wrap listeners with timeout cancellation logic,
+ * and fetch workflow states after timeouts.
+ */
+public class WorkflowTimeoutUtility {
+
+    private static final Logger logger = LogManager.getLogger(WorkflowTimeoutUtility.class);
+    private static final TimeValue MIN_TIMEOUT_MILLIS = TimeValue.timeValueSeconds(0);
```

Comment:
This works but you're just doing a fancy wrapper for 0.

Can probably delete this constant and just use 0 where you are calling `MIN_TIMEOUT_MILLIS.millis()`.